packages/select/src/ListMenu.tsx

```diff
@@ -44,18 +48,33 @@ const ListMenu = React.forwardRef<HTMLUListElement, ListMenuProps>(
     forwardedRef,
   ) {
     const { mode, size, disabled, open } = useContext(SelectContext);
-
     const colorSet = colorSets[mode];
     const sizeSet = sizeSets[size];
 
     const ref = useForwardedRef(forwardedRef, null);
 
     const viewportSize = useViewportSize();
 
-    const maxHeight =
-      viewportSize === null || ref.current === null
-        ? 0
-        : viewportSize.height - ref.current.getBoundingClientRect().top - 10;
+    const maxHeight = useMemo(() => {
+      if (viewportSize && ref.current && referenceElement.current) {
+        const {
+          top: triggerTop,
+          bottom: triggerBottom,
+        } = referenceElement.current.getBoundingClientRect();
+
+        // Find out how much space is available above or below the trigger
+        const safeSpace = Math.max(
+          viewportSize.height - triggerBottom,
+          triggerTop,
+        );
+
+        // if there's more than enough space, set to MAX_MENU_HEIGHT
+        // otherwise fill the space available
+        return Math.min(MAX_MENU_HEIGHT, safeSpace - MENU_MARGIN);
+      } else {
```

Comment:
Small nit, but one standard we've applied fairly consistently in LeafyGreen is to use happy-path coding techniques. So rather than what you have here, it would move the else block return out of the conditional:

```js
const maxHeight = useMemo(() => {
    if (viewportSize && ref.current && referenceElement.current) {
      /* code here */
      return Math.min(MAX_MENU_HEIGHT, safeSpace - MENU_MARGIN);
    } 

    return MAX_MENU_HEIGHT;
}, [ref, referenceElement, viewportSize]);
```

Reply:
Makes sense! I'll make that update